src/pages/Navgurukul/merakiAdmission/index.js

```diff
@@ -0,0 +1,211 @@
+      const response = await axios.get(dataURL, {
+        params: {
+          ngCallType: "getEnrolmentKey",
+          From: mobile,
+          partner_id: partnerId ? partnerId : null,
+        },
+      });
+      setEnrolmentKey(response.data.key);
+      const params = {
+        firstName: userDetails.firstName,
+        middleName: userDetails.middleName,
+        lastName: userDetails.lastName,
+        mobileNumber: userDetails.mobileNumber,
+      };
+      const queryString = Object.keys(params)
+        .map((filter) => {
+          if (params[filter]) {
+            return `${filter}=${params[filter]}`;
+          }
+          return null;
+        })
+        .filter((item) => item)
+        .join("&");
+
+      const url = `${testUrl}${enrolmentKey}?${queryString}`;
+
+      window.open(url, "_blank");
+      return response;
+    } catch (e) {}
+  };
+
+  const giveTest = async () => {
+    if (
+      !userDetails.firstName ||
+      !userDetails.lastName ||
+      !userDetails.mobileNumber
+    ) {
+      toast.error(
+        "To attempt the test, it is compulsory to enter your First Name, Last Name and Mobile Number. Middle Name is optional, you can choose not to enter.",
+        {
+          position: toast.POSITION.BOTTOM_RIGHT,
+        }
+      );
+      return;
+    }
+    if (userDetails.mobileNumber.toString().length !== 10) {
+      toast.error("Please give 10 digits of the mobile number.", {
+        position: toast.POSITION.BOTTOM_RIGHT,
+      });
+      return;
+    }
+    await isDuplicate();
+  };
+
+  const isDuplicate = () => {
+    const { firstName, middleName, lastName, mobileNumber } = userDetails;
+    axios
+      .get(`${process.env.REACT_APP_CHANAKYA_BASE_URL}/check_duplicate`, {
+        params: {
+          Name: firstName.concat(" ", middleName, lastName),
+          Number: mobileNumber,
+        },
+      })
+      .then(async (data) => {
+        const response = data.data.data;
+        const stage = response.pendingInterviewStage;
+        const url = `https://admissions.navgurukul.org/check_duplicate/Name=${firstName}${middleName}${lastName}&Number=${mobileNumber}&Stage=${stage}`;
+        if (response.alreadyGivenTest) {
+          window.open(url, "_blank");
+        } else {
+          generateTestLink();
+        }
+      });
+  };
+
+  const changeHandler = (e) => {
+    setUserDetails({ ...userDetails, [e.target.name]: e.target.value });
+  };
+
+  return (
+    <div className="admission">
+      <div className="left-section">
+        <YouTube className={"video-alumni"} videoId={`vuSwndj5cbs`} />;
+        <div className="video-label">Experience of NG Alumni & Graduates</div>
+      </div>
+      <div className="right-section">
+        <div className="test-form">
+          <h2 className="scholarship-test">
+            Software Engineering Scholarship Test
+          </h2>
+          <div className="input-form">
```

Comment:
For better accessibility (See: https://reactjs.org/docs/accessibility.html#accessible-forms), you can use the [`<label>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label) element for label text for each of the form inputs instead of `<span>`.  If you surround the `<input>` elements with `<label>` and `</label>`, the person will be able to click on the label (First Name, Middle Name, Last Name, or Mobile Number) and the inputs will be put into focus.

Reply:
`   <label>First Name</label>
                <input
                  type="text"
                  placeholder="Abhi..."
                  value={userDetails.firstName}
                  name="firstName"
                  onChange={changeHandler}
                  required
                />`
               I did like this because wrapping `<input>` with `<label>` caused to bring the label and field on the same line.

Reply:
In that case, to keep a connection between the `<label>` and the `<input>`, you can give the `input` an `id` and use the [`for`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label#attr-for) attribute.  However, because `for` is a keyword in JavaScript, you do this with `htmlFor` in React.  So for example, you could add an `id` of `firstName` to the input and then the label would be `<label htmlFor = "firstName">`.